### ISSUE TITLE
Revise GL CI parallelization and dependencies

### ISSUE THREAD

**Issue by thisismydesign** [c-hive/guides#22]
So if I understand it correctly we should have the following stages and jobs:
```
stages:
  - verification
  - deploy

lint:
  stage: verification

test:
  stage: verification

apiary-validate:
  stage: verification

code-quality:
  stage: verification

heroku-deploy:
  stage: deploy

gl-pages-deploy:
  stage: deploy

apiary-deploy:
  stage: deploy
```

This would result in maximum parallelization while making sure deploys only run if verifications are successful.

**Comment by gomorizsolt** [comment]
That's it, all the jobs under the `verification` stage run in parallel, whereas the `deploy` runs completely separately.

**Comment by gomorizsolt** [comment]
Resolved by #23.
